usaspending_api/search/elasticsearch/filters/naics.py

```diff
@@ -0,0 +1,103 @@
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.search.elasticsearch.filters.filter import _Filter, _QueryType
+from elasticsearch_dsl import Q as ES_Q
+
+
+class NaicsCodes(_Filter):
+    underscore_name = "naics_codes"
+
+    @classmethod
+    def generate_elasticsearch_query(cls, filter_values, query_type: _QueryType) -> ES_Q:
+        # legacy functionality permits sending a single list of naics codes, which is treated as the required list
+        if isinstance(filter_values, list):
+            require = filter_values
+            exclude = []
+        elif isinstance(filter_values, dict):
+            require = filter_values.get("require") or []
+            exclude = filter_values.get("exclude") or []
+        else:
+            raise InvalidParameterException(f"naics_codes must be an array or object")
+
+        requires = [str(code) for code in require]
+        exclude = [str(code) for code in exclude]
+
+        return ES_Q("query_string", query=cls._query_string(requires, exclude), default_field="naics_code")
```

Comment:
❗️ This needs to be changed to `naics_code.keyword` yes? 

Please use the keyword sub-fields or fields with keyword types as much as possible in Advanced Search filters.